lib/mix/lib/mix/tasks/format.ex

```diff
@@ -454,7 +460,14 @@ defmodule Mix.Tasks.Format do
 
     for file <- files do
       if file == :stdin do
-        {file, &elixir_format(&1, [file: "stdin"] ++ formatter_opts)}
+        if stdin_filename = Keyword.get(opts, :stdin_filename) do
+          {formatter, _opts} =
+            find_formatter_and_opts_for_file(stdin_filename, {formatter_opts, subs})
+
+          {file, formatter}
+        else
+          {file, &elixir_format(&1, [file: "stdin"] ++ formatter_opts)}
+        end
```

Comment:
```suggestion
        stdin_filename = Keyword.get(opts, :stdin_filename, "stdin.exs")

        {formatter, _opts} =
            find_formatter_and_opts_for_file(stdin_filename, {formatter_opts, subs})

        {file, formatter}
```

WDYT?

Reply:
What as the error before? Was it "nofile"? If it was "nofile", I think we can also just "nofile" instead of "stdin.exs".

Reply:
From https://github.com/elixir-lang/elixir/blob/496a02c741c2dc82b4d8f27b5102e4595c4db1c4/lib/mix/test/mix/tasks/format_test.exs#L639 -- it was:

```
stdin:1:13: syntax error before: '='
```

So the change is `stdin.exs` instead of `stdin`. I'm not sure if there are tools out there that wrap `mix format -` currently and expect the exact string `stdin` when parsing the message.

Reply:
I see. Let's try "stdin", if it is not enough, going with "stdin.exs" is totally fine IMO.

Reply:
Cool, just pushed a change that includes a special-case in `find_formatter_for_file` -- let me know if that's what you had in mind

Reply:
Oh, I see. So let's rename it to `stdin.exs` indeed and remove the special case. Sorry for the back and forth.